Changes to allow output to be a gzip file

kiwi1969 opened the pull request:
Changes to allow output to be a gzip file (if the filename ends with ".gz")

kiwi1969 committed:
Update README.md

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Update pom.xml

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Handle gzip output files

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Handle gzip output files

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Handle gzip output files

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Handle gzip output files

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

alexgubanow commented:
please target "release-5.0.1" branch first
do the change less invasive, like removal try{} is no go
branch name has to be created from issue, so it will be reflected in issue log

kiwi1969 committed:
Update README.md

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Update pom.xml

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 committed:
Update TerseDecompress.java

Restored try clause.
Increased GZIP buffer size for effciency and for CryptoCards that only get used if buffer > = 8192 bytes
Set GZIP flush option to prevent corruption of file on last write

Signed-off-by: Russell Shaw <69813534+kiwi1969@users.noreply.github.com>

kiwi1969 commented:
Updated code as suggested. Added buffer=8192 and syncflush parms to open of gzip file, so last record is written , and file is not corrupt. 
Not sure how to create branch from issue - I cannot find such an option

alexgubanow commented on src/main/java/org/openmainframeproject/tersedecompress/TerseDecompress.java:
```diff
@@ -95,15 +99,23 @@ else if (outputFileName == null)
     	}
 
+		System.out.println("Attempting to decompress input file (" + inputFileName +") to output file (" + outputFileName +")");
+
```
code below has duplicate lines, which may lead to errors later. more cleaner solution would be to create an object of FileOutputStream, while GZIPOutputStream is overload of such.
```suggestion
var outputStream = null;
		try {
		if (outputFileName.endsWith(".gz")) {
			outputStream = new GZIPOutputStream(new FileOutputStream(outputFileName), 8192, true);
		}
		else {
			outputStream = new FileOutputStream(outputFileName);
		}
		}
			try (TerseDecompresser outputWriter = TerseDecompresser.create(new FileInputStream(inputFileName), outputStream ))
			{	 
				outputWriter.TextFlag = textMode;
				outputWriter.decode();
			}
```
suggested code is just to describe the idea, please test it
